tests/unit/pipeline/test_dag.py

```diff
@@ -504,15 +504,15 @@ def test_dag_from_dict(self, dummy_step_1: "Step", dummy_step_2: "Step") -> None
         assert "dummy_step_2" in new_dag.G["dummy_step_1"]
 
     def test_dag_from_dict_errored_without_pipeline(
-        self, dummy_step_1: "Step", dummy_step_2: "Step"
+        self, caplog, dummy_step_1: "Step", dummy_step_2: "Step"
```

Comment:
Maybe it's nice to also add the type-hint here

```suggestion
        self, caplog: pytest.LogCaptureFixture, dummy_step_1: "Step", dummy_step_2: "Step"
```